docs/ndcubesequence.rst

```diff
@@ -0,0 +1,295 @@
+along the sequence axis from one cube to the next both represent
+movement in time.  The difference is simply size of the steps.
+Therefore it can be said that the 0th axis of the sub-cubes is common
+to the sequence. 
+
+To define a common axis, set the kwarg during intialization of
+the `~ndcube.NDCubeSequence` to the desired data axis number::
+
+  >>> my_sequence = NDCubeSequence([my_cube0, my_cube1, my_cube2],
+  ...                              meta=my_sequence_metadata, common_axis=0)
+
+Defining a common axis enables the full range of the
+`~ndcube.NDCubeSequence` features to be utilized including
+`ndcube.NDCubeSequence.plot`,
+`ndcube.NDCubeSequence.common_axis_extra_coords`, and 
+`ndcube.NDCubeSequence.index_as_cube`. See following sections for
+more details on these features.
+
+Dimensions
+----------
+
+Analagous to `ndcube.NDCube.dimensions`, there is also a
+`~ndcube.NDCubeSequence.dimensions` property for easily inspecting the
+size and shape of an `~ndcube.NDCubeSequence` instance::
+
+  >>> my_sequence.dimensions
+  SequenceDimensionPair(shape=(3, <Quantity 3.0 pix>, <Quantity 4.0 pix>, <Quantity 5.0 pix>), axis_types=('Sequence Axis', 'HPLN-TAN', 'HPLT-TAN', 'WAVE'))
+
+
+Like NDCube it returns a named tuple with a "shape" and "axis_types"
+where the values of the 0th sub-cube are returned as
+`astropy.units.Quantity` and `str` objects, respectively.  In addition
+however, another dimension is return at the start of the named tuple.
+Its value is an `int` (not a Quantity) giving the number of sub-cubes
+in the sequence.  Since this is not a WCS axis, its axis type is given
+the label ``'Sequence Axis'``.
+
+As the dimensions property returns a named tuple, the shape and axis
+types can be accessed directly::
+
+  >>> my_sequence.dimensions.shape
+  (3, <Quantity 3.0 pix>, <Quantity 4.0 pix>, <Quantity 5.0 pix>)
+  
+  >>> my_sequence.dimensions.axis_types
+  ('Sequence Axis', 'HPLN-TAN', 'HPLT-TAN', 'WAVE')
+
+Slicing
+-------
+As with `~ndcube.NDCube`, slicing an `~ndcube.NDCubeSequence` using
+the standard slicing API simulataneously slices the data arrays, WCS
+objects, masks, uncertainty arrays, etc. in each relevant sub-cube.
+For example, say we have three NDCubes in an `~ndcube.NDCubeSequence`,
+each of shape (10, 20, 30).  Say we want to obtain a region of
+interest between the 1st and 2nd pixels (inclusive) in the 2nd
+dimension and 1st and 3rd pixels (inclusive) in the 3rd dimension of
+the 0th slice along the 0th axis in only the 1st (not 0th) and 2nd
+sub-cubes in the sequence. This would a cumbersome slicing operation
+if treating the sub-cube independently. (This would be made even worse
+without the power of `~ndcube.NDCube` where the data arrays, WCS
+objects, masks, uncertainty arrays, etc. would all have to be sliced
+independently!) However, with `~ndcube.NDCubeSequence` this becomes as
+simple as indexing a single array::
+
+  >>> ######### Due to a bug, this currently doesn't work! #########
```

Comment:
I would hope the doctests would fail (meaning you need to skip it for now)

Reply:
Done